extensions/interactions/NumberWithUnits/directives/number-with-units-rules.service.ts

```diff
@@ -50,11 +50,20 @@ export class NumberWithUnitsRulesService {
     var answerString = answerObject.toMathjsCompatibleString();
     var inputsString = inputsObject.toMathjsCompatibleString();
 
-    var answerList = this.unitsObjectFactory.fromRawInputString(
-      answerString).toDict();
-    var inputsList = this.unitsObjectFactory.fromRawInputString(
-      inputsString).toDict();
-    return this.utilsService.isEquivalent(answerList, inputsList);
+    var answerList: Partial<NumberWithUnitsAnswer> = this.unitsObjectFactory
+      .fromRawInputString(
+        answerString).toDict();
+    var inputsList: Partial<NumberWithUnitsAnswer> = this.unitsObjectFactory
+      .fromRawInputString(
+        inputsString).toDict();
+
+    delete answerList.units;
```

Comment:
Please help me to understand why to delete the units.

Reply:
the utilsService.isEquivalent function strictly compare the input objects and in this case we only want to strictly compare the values not there units.


So we are using utilsService.isEquivalent to strictly compare value and isEquivalentTo function for units.

Reply:
> in this case we only want to strictly compare the values not there units.

Yes, we don't compare strictly, but imagine a case where the unit is "J/s" and the user provides "K/s"  so how do we handle this condition (please point out the code)?

One solution I can think of is: can we convert the input into some standard form and perform strict checking? Because deleting the unit from the input is not a good approach, in my opinion,

Let me know if you need some more clarification.
Thanks.